presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -299,10 +299,10 @@ public Expression getJoinCriteria(Join join)
     public void recordSubqueries(Node node, ExpressionAnalysis expressionAnalysis)
     {
         NodeRef<Node> key = NodeRef.of(node);
-        this.inPredicatesSubqueries.putAll(key, expressionAnalysis.getSubqueryInPredicates());
-        this.scalarSubqueries.putAll(key, expressionAnalysis.getScalarSubqueries());
-        this.existsSubqueries.putAll(key, expressionAnalysis.getExistsSubqueries());
-        this.quantifiedComparisonSubqueries.putAll(key, expressionAnalysis.getQuantifiedComparisons());
+        this.inPredicatesSubqueries.putAll(key, NodeRefCollections.toIdentitySet(expressionAnalysis.getSubqueryInPredicates()));
```

Comment:
I would just squash it with previous commit: `Remove identity-collection use in ExpressionAnalyzer`. I don't see the benefit of intermittent commit.